File: src/main/Java/boj_step4/boj10818.java
import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

public class boj10818 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int[] list = new int[n];
        for (int i = 0; i < n; i++) list[i] = Integer.parseInt(tokenizer.nextToken());
        int max = Arrays.stream(list).max().getAsInt();
        int min = Arrays.stream(list).min().getAsInt();
        writer.append(min + " " + max);
        writer.close();
    }
}

File: src/main/kotlin/boj10818.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.math.min

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine()!!.toInt()
    val arr = r.readLine()!!.split(" ").map { it.toInt() }
    w.write("${arr.minOrNull()} ${arr.maxOrNull()}")
    w.close()
}